imap/mailbox.c

```diff
@@ -7942,8 +7990,10 @@ EXPORTED const message_t *mailbox_iter_step(struct mailbox_iter *iter)
 {
     if (mailbox_wait_cb) mailbox_wait_cb(mailbox_wait_cb_rock);
 
-    for (iter->recno++; iter->recno <= iter->num_records; iter->recno++) {
```

Comment:
This feels complicated when we could PROBABLY just use a iter_step_backwards API rather than doing two tests every loop

Reply:
I did it this way so we didn't have to replicate code or make function call to run the common code.
We are already doing 6-8 tests per loop as it is, but if you still have a problem with it, I'll rewrite it.